メッセージ送信機能実装

hidetoshi1230 opened the pull request:
WHAT
メッセージ送信機能の実装

WHY
サービスに必要なため。

hidetoshi1230 committed:
メッセージ送信機能実装1（ルーティング、コントローラーの設定）

ルーティングを設定する
該当するアクションをコントローラに定義する

hidetoshi1230 committed:
メッセージ送信機能実装2（送信機能、メッセージ表示）

・メッセージ送信機能を実装する
フラッシュメッセージ、アラート、画像アップデート、gitignore、CarrierWaveインストール
・グループにメッセージを表示する

TOMARUMARU commented on app/controllers/messages_controller.rb:
```diff
@@ -1,3 +1,24 @@
+    @groups = current_user.groups
+    @group  = Group.find(params[:group_id])
+    @message = Message.new
+    @messages = Message.where(group_id: params[:group_id]).order("created_at ASC")
```
こちらでも取得できますが、アソシエーションを利用して取得したほうが、より簡単に取得できます。

hidetoshi1230 committed:
メッセージ送信機能実装3（修正）

こちらでも取得できますが、アソシエーションを利用して取得したほうが、より簡単に取得できます。

TOMARUMARU commented on app/controllers/messages_controller.rb:
```diff
@@ -1,3 +1,24 @@
+    if @message.save
+      redirect_to group_messages_path
+    else
+      redirect_to group_messages_path, alert: "メッセージを入力してください。"
```
こちらはredirectではなくrenderにて実装してください。

TOMARUMARU commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,52 @@
+          = link_to new_group_path do
+            %i.fa.fa-pencil-square-o
+        %li
+          = link_to edit_user_path do
```
こちらエラー出ていませんか？

TOMARUMARU commented on public/.gitignore:
```diff
@@ -0,0 +1 @@
+*.jpg
```
jpgを指定するのではなく、public/uploads以下をignoreするようにしましょう

hidetoshi1230 committed:
メッセージ送信機能実装４（修正）

こちらはredirectではなくrenderにて実装してください。
こちらエラー出ていませんか？
jpgを指定するのではなく、public/uploads以下をignoreするようにしましょう

hidetoshi1230 committed:
メッセージ送信機能実装5（細かい部分を修正する）

サイドバーのグループ部分に最新のメッセージを表示

kouuuki commented on app/controllers/messages_controller.rb:
```diff
@@ -1,3 +1,28 @@
 class MessagesController < ApplicationController
 
+  def index
+    @group  = Group.find(params[:group_id])
```
同じ処理はメソッドにまとめてbefore_actionで呼び出しましょう！

kouuuki commented on app/controllers/messages_controller.rb:
```diff
@@ -1,3 +1,28 @@
+
+  def create
+    @message = Message.new(message_params)
+    @group  = Group.find(params[:group_id])
```
同じ処理はメソッドにまとめてbefore_actionで呼び出しましょう！

hidetoshi1230 committed:
メッセージ送信機能実装6（修正）

同じ処理はメソッドにまとめてbefore_actionで呼び出しましょう！

kouuuki commented:
LGTMです！
